plugin/src/main/java/io/jenkins/plugins/remotingkafka/GlobalKafkaConfiguration.java

```diff
@@ -211,6 +225,44 @@ public FormValidation doTestBrokerConnection(@QueryParameter("brokerURL") final
         }
     }
 
+    @RequirePOST
+    public FormValidation doTestKubernetesConnection(
+            @QueryParameter("kubernetesUrl") String serverUrl,
+            @QueryParameter("kubernetesCredentialsId") String credentialsId,
+            @QueryParameter("kubernetesCertificate") String serverCertificate,
+            @QueryParameter("kubernetesSkipTlsVerify") boolean skipTlsVerify,
+            @QueryParameter("kubernetesNamespace") String namespace
+    ) {
+        Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+
+        try (KubernetesClient client = new KubernetesFactoryAdapter(serverUrl, namespace,
+                Util.fixEmpty(serverCertificate), Util.fixEmpty(credentialsId), skipTlsVerify
+        ).createClient()) {
+            client.pods().list();
```

Comment:
what is the purpose of listing pods here?

Reply:
I borrowed that line from kubernetes-plugin. I'm not sure why they wrote it but maybe they want to make sure that not only the client connection can be established but it also has to be functional (by try calling pod list API). Maybe I should write a comment so future code-readers won't be confused?